components/HeroIcon/HeroIcon.tsx

```diff
@@ -16,11 +16,11 @@ export const HeroIcon = ({
   className,
   ...props
 }: HeroIconProps) => {
-  const Icon = styles.iconMap[icon || ''];
+  const Icon = icon ? styles.iconMap[icon] : undefined;
 
   // Set default base style so icon has reasonable size if used out of the box
   // noBaseStyle boolean allows for user to not attach any base styles if needed
-  const baseStyle = noBaseStyle ? '' : styles.iconBaseStyle[icon || ''] || styles.iconBaseStyle.default;
+  const baseStyle = noBaseStyle ? '' : styles.iconBaseStyle[icon as styles.IconBaseStyleType] || styles.iconBaseStyle.default;
```

Comment:
Typecasting here is totally cheating. The icon param is optional and could be undefined. It would be way better to either require the prop or add some type guards.

Reply:
Yeah I typed everything that I could already but it was still giving me error. Let me see if there's any other way I could avoid the typecasting.